raft/src/main/java/org/apache/kafka/raft/BatchReader.java

```diff
@@ -60,19 +61,21 @@
     @Override
     void close();
 
-    class Batch<T> {
+    final class Batch<T> implements Iterable<T> {
```

Comment:
@hachikuji given that we use this type in `SnapshotReader<T>` I am okay moving this type to something like `o.a.k.r.Batch` instead of `o.a.k.r.BatchReader.Batch`.